airflow/utils/sqlalchemy.py

```diff
@@ -214,6 +214,9 @@ def _validate_commit(self, _):
         if self.expected_commit:
             self.expected_commit = False
             return
+        # Don't raise error if session is clean (useful when just querying from DB)
+        if self.session._is_clean():  # pylint: disable=protected-access
```

Comment:
This can't be here - even if the session is clean, if we allow a COMMIT the lock will be released

Reply:
Yeah, I tried with `session.execute("update dag set is_active=False where dag_id='example_branch_operator'")` where `session._is_clean()` is `True` -- so this logic does not work -- going to close the PR